### DISCUSSION TITLE
audioread.exceptions.NoBackendError

### DISCUSSION THREAD

**Discussion by becausebirds** [mcguirepr89/BirdNET-Pi#746] [Q&A]
My station stopped analyzing as of five hours ago and I see this in the logs. Any advice?

> 18:18:56---    sig, rate = librosa.load(path, sr=sample_rate, mono=True, res_type='kaiser_fast')
> 18:18:56---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/util/decorators.py", line 88, in inner_f
> 18:18:56---    return f(*args, **kwargs)
> 18:18:56---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/core/audio.py", line 170, in load
> 18:18:56---    y, sr_native = __audioread_load(path, offset, duration, dtype)
> 18:18:56---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/core/audio.py", line 226, in __audioread_load
> 18:18:56---    reader = audioread.audio_open(path)
> 18:18:56---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/audioread/__init__.py", line 132, in audio_open
> 18:18:56---    raise NoBackendError()
> 18:18:56---audioread.exceptions.NoBackendError

It's still recording and producing new files, but the only thing that ever shows in the log is the above message. I've tried clicking on "update" several times, stopping and restarting core services, and cutting and restoring power.

**Comment by ehpersonal38** [comment]
Can you try making sure the following 2 packages are installed?

`sudo apt-get install libav-tools ffmpeg`

**Reply by becausebirds** [reply]
@ehpersonal38 Thanks for the suggestion. Earlier today I was just doing a combination of toggling the model, clicking update, and rebooting device/services and it started working again. It just seems to be analyzing the backlog WAY slower than normal currently... Even when I try to watch the log, it shows a few clips get processed the terminal loses connection.

**Reply by ehpersonal38** [reply]
Can you navigate to Tools -> System Info and provide screenshots or copy and paste the text for these 3 areas of interest?

![chrome_O8vn7O3e95](https://user-images.githubusercontent.com/103586016/219974268-40c7d129-c64f-45a5-8658-5a5f705170e8.png)
![chrome_xTjuVj00K5](https://user-images.githubusercontent.com/103586016/219974275-9311b574-cc65-4334-8041-26bee4353fca.png)
![chrome_cA88k3YDsQ](https://user-images.githubusercontent.com/103586016/219974278-cf766136-9cd2-4a6c-8c9c-db6a81bfd0fc.png)

**Reply by becausebirds** [reply]
<img width="1157" alt="Screenshot 2023-02-19 at 2 51 38 PM" src="https://user-images.githubusercontent.com/110354170/219974513-f6479d87-0164-4b99-9f1e-53cac7f2097f.png">
<img width="569" alt="Screenshot 2023-02-19 at 2 51 48 PM" src="https://user-images.githubusercontent.com/110354170/219974515-95ef48a7-ec35-4eed-b668-35473f648cd3.png">
<img width="455" alt="Screenshot 2023-02-19 at 2 52 19 PM" src="https://user-images.githubusercontent.com/110354170/219974517-d8acf14f-d55e-4541-9b74-062a986f2257.png">
 Here ya go! Nothing has changed with its power source recently.

**Reply by ehpersonal38** [reply]
So it looks like your Pi is receiving too little voltage. I'd recommend replacing your power supply with one that can supply 5.1 volts DC at 2.5 amps.

Also since you're on a Pi 3B+ it will be worth increasing your swap space with these commands - https://github.com/mcguirepr89/BirdNET-Pi/wiki/RPi0W2-Installation-Guide#how-to-install-on-the-raspberry-pi-0-w-2

**Reply by becausebirds** [reply]
@ehpersonal38 While I hunt down a power supply that fits in the box, is it safe to increase the swap space in the mean time?

**Reply by ehpersonal38** [reply]
Yep you're free to increase the swap space at any time. I should just add it to the installer to do automatically at this point.

**Reply by becausebirds** [reply]
@ehpersonal38 I've increased the swap memory and swapped the power supply with a brand new one rated for 5V/3A on USB-A. For some reason, the system info still shows it's undervoltage, unless I'm reading it wrong.

<img width="380" alt="Screenshot 2023-02-19 at 3 46 30 PM" src="https://user-images.githubusercontent.com/110354170/219977174-8b77ce90-bcb3-4396-8834-1901997ec14a.png">

**Reply by ehpersonal38** [reply]
Hmm, is it possible for you to try with another cable? Preferrably the highest quality cable you can find, like the ones that come with smartphones.

**Comment by becausebirds** [comment]
The same issue happened again this weekend. It seems it was getting hung up on a particular extracted file. Deleting it and restarting services seems to have fixed the issue. Not sure what is causing it to hang up in the first place.

**Reply by rbrtgg** [reply]
Any idea as to what that file might have been and where I can find what might be causing the error?  I have been running this application for about a year and last night I got this:
14:14:38--- sig, rate = librosa.load(path, sr=sample_rate, mono=True, res_type='kaiser_
14:14:38--- File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/util/decorators.
14:14:38--- return f(*args, **kwargs)
14:14:38--- File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/core/audio.py",
14:14:38--- y, sr_native = __audioread_load(path, offset, duration, dtype)
14:14:38--- File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/librosa/core/audio.py",
14:14:38--- with audioread.audio_open(path) as input_file:
14:14:38--- File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/audioread/__init__.py",
14:14:38--- raise NoBackendError()
14:14:38---audioread.exceptions.NoBackendError

There was an outstanding update which I have since done - otherwise it has been running fine for ages.  Reading the comments above, I don't see any issues with power or the like (I did try switching the power supply from my other pi - no change),  I shut everything down and restarted.

Any ideas as to what I should try?   Thanks

**Reply by rbrtgg** [reply]
Somehow it stayed in this state for one day and then the next day it started back as normal.  Beyond me but I'm happy it's back.
